### DISCUSSION TITLE
Worker and Requeuer exiting for no good reason?

### DISCUSSION THREAD

**Discussion by gerricom** [postalserver/postal#2360] [Help with using Postal]
Hello everybody!
I'm having issues with our postal server at the moment as the worker and requeuer are exiting for no obious reason. I read some other discussions and issues here, but they are not helping at the moment. 

For those exits I don't see any pattern. Sometimes it's quiet for weeks, and then we have this issue two times in 48 hours.

This is what I checked so far:

- Ressource consumption in general: One of the hints in some other threads were, that services are exiting because of hitting ressource limits. Memory consumption is around 30 % on this host (running postal alone), CPU has no relevant peaks and the disk is at 40 % full
- Logs: worker and requeuer don't show *any* log entries. Is this normal? I expected some "worker starting..." messages... well?
- rabbitmq's logs: I'm running rabbitmq as recommended in the documentation. I'm seeing something like this there:

```
2023-02-22 05:04:08.710 [warning] <0.8077.1> closing AMQP connection <0.8077.1> (172.17.0.1:49686 -> 172.17.0.2:5672, vhost: 'postal', user: 'postal'):
client unexpectedly closed TCP connection
2023-02-22 05:04:08.712 [info] <0.10712.6> Closing all channels from connection '172.17.0.1:49686 -> 172.17.0.2:5672' because it has been closed
2023-02-22 05:04:20.638 [warning] <0.8090.1> closing AMQP connection <0.8090.1> (172.17.0.1:49690 -> 172.17.0.2:5672, vhost: 'postal', user: 'postal'):
client unexpectedly closed TCP connection
2023-02-22 05:04:20.638 [info] <0.10718.6> Closing all channels from connection '172.17.0.1:49690 -> 172.17.0.2:5672' because it has been closed
```

To me this sounds like the client is exiting and that there is no issue with rabbitmq itself.

What are other things to check?

I'm thinking of running a cron job doing `postal start` every hour or so but this is obviously not the solution rather a hack to keep things running...

**Comment by willpower232** [comment]
Were you trying `postal logs worker`? There should be something. Otherwise I might suggest enabling sentry to get more detailed reporting on problems.

We know those two fail if there is a connection issue to the RabbitMQ server. If RabbitMQ is being hosted elsewhere then it sounds like your memory is fine but maybe there are network issues somehow?

**Reply by gerricom** [reply]
> Were you trying postal logs worker? There should be something. 

It looks like this:

```
root@lessing:~# postal logs worker
Attaching to postal_worker_1
root@lessing:~# 
```

> We know those two fail if there is a connection issue to the RabbitMQ server. If RabbitMQ is being hosted elsewhere then it sounds like your memory is fine but maybe there are network issues somehow?

RabbitMQ is running local on the same machine as postal. So a network issue should be quite impossible...

> Otherwise I might suggest enabling sentry to get more detailed reporting on problems. 

I didn't know that this is possible. I added the exception_url to our Sentry installation and will have a look...

**Reply by willpower232** [reply]
the only other thing you could do is have a look in the server log files to see if there have been any OOMkills or problems with the services in case memory ran out briefly enough to cause issue but not long enough to crop up on monitoring or whatever
